### PR TITLE
feat(levm): initialize cache with sender, recipient and coinbase accounts

### LINKED ISSUE #1088: LEVM: Substate and Cache should be initialized with the right values

### PR DESCRIPTION
**Pull request by JereSalo**
**Motivation**

**Description**

As stated in [evm.codes](https://www.evm.codes/about#access_list), this PR initializes the cache with the corresponding addresses. (Without implementing yet the concept of an Access List transaction)

Notes: Our `Cache` struct replaces the need for a `Substate` that keeps warm addresses and storage slots because we know that if an account or storage slot is in our cache then it is already warm.

Closes #1088

### CODE REVIEW

File: crates/vm/levm/src/account.rs [hunk 1]
```diff
@@ -48,6 +48,14 @@ impl Account {
         }
     }
 
+    /// New account with info and empty storage
+    pub fn new_with_info(info: AccountInfo) -> Self {
+        Self {
+            info,
+            storage: HashMap::new(),
+        }
+    }
```
Comment (ilitteri) [inline crates/vm/levm/src/account.rs:57]: As an exercise, replace this implementation with `From<AccountInfo> for Account`.

**Review by ilitteri — CHANGES_REQUESTED** [review]
The PR looks good! Left one comment as an exercise. It similar cases, it is always preferrable to implement the `From` trait and `TryFrom` if the conversion could fail.

File: crates/vm/levm/src/account.rs (hunk 1 above)
Comment (JereSalo) [reply crates/vm/levm/src/account.rs]: Done!

Comment (JereSalo) [reply crates/vm/levm/src/account.rs]: https://github.com/lambdaclass/lambda_ethereum_rust/pull/1114/commits/4d0413a2daaa2908c876fc15e769e4f056ff45f9

**Review by ilitteri — APPROVED** [review]
LGTM. We can merge after resolving the conflicts.

**Pull request merged**